remote/mdb_client_test.go

```diff
@@ -0,0 +1,109 @@
+package remote
+
+import (
+	"context"
+	"runtime"
+	"syscall"
+	"testing"
+
+	"github.com/mongodb/jasper"
+	"github.com/mongodb/jasper/options"
+	"github.com/mongodb/jasper/testutil"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// Note: these tests are largely copied directly from the top level package into
+// this package to avoid an import cycle.
+
+func TestWireManager(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	factory := func(ctx context.Context, t *testing.T) Manager {
+		mngr, err := jasper.NewSynchronizedManager(false)
+		require.NoError(t, err)
+
+		client, err := makeTestServiceAndClient(ctx, mngr)
+		require.NoError(t, err)
+		return client
+	}
+
+	for _, modify := range []struct {
+		Name    string
+		Options testutil.OptsModify
+	}{
+		{
+			Name: "Blocking",
+			Options: func(opts *options.Create) {
+				opts.Implementation = options.ProcessImplementationBlocking
+			},
+		},
+		{
+			Name: "Basic",
+			Options: func(opts *options.Create) {
+				opts.Implementation = options.ProcessImplementationBasic
+			},
+		},
+		{
+			Name:    "Default",
+			Options: func(opts *options.Create) {},
+		},
+	} {
+		t.Run(modify.Name, func(t *testing.T) {
+			for _, test := range AddBasicClientTests(
+				ClientTestCase{
+					Name: "RegisterIsDisabled",
```

Comment:
These tests are currently separated run in just the mdb/grpc tests, but they should probably just run for all remote interfaces. A lot of these tests should probably be run in the common test suite rather than in service-specific tests. Can you add a TODO to remove duplicate tests?

Reply:
merged and removed everything in this file. I'll open a ticket to deduplicate the rest tests which are different.